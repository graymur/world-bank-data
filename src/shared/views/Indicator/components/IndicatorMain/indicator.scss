@import "~style/lib";

.indicator {
    position: relative;

    &__years {
        display: block;
        margin-top: 2rem;
        font-size: 1.8rem;

        &__item {
            display: inline-block;
            margin-right: 2rem;

            &._active {
                color: black;
                text-decoration: none;
                pointer-events: none;
            }
        }
    }
}