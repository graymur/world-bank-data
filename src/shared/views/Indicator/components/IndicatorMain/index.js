import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import getPageTitle from 'shared/utils/getPageTitle';
import Loader from 'shared/components/Loader';
import IndicatorData from '../IndicatorData';
// import SuggestData from '../SuggestData';
import classnames from 'classnames';
import IndicatorDataByYearProvider from 'shared/providers/IndicatorDataByYear';
import './indicator.scss';

export default class IndicatorMain extends React.Component {
	static propTypes = {
		loading: PropTypes.bool,
		years: PropTypes.array,
		indicator: PropTypes.object,
		changeYear: PropTypes.func,
		currentYear: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
	};

	constructor(props) {
		super(props);
		this.handleYearChange = this.handleYearChange.bind(this);
	}

	handleYearChange(event) {
		if (event.target.value) {
			this.props.changeYear(Number(event.target.value));
		}
	};

	rendersYears() {
		const {years, currentYear} = this.props;

		const defaultSelectValue = isNaN(Number(currentYear)) ? '' : Number(currentYear);
		return (
			<div className='indicator__years'>
				<select key={currentYear} id='year' className='styled-select' onChange={this.handleYearChange} defaultValue={defaultSelectValue}>
					<option value='' hidden>Select year</option>
					{years.map(year => (
						<option key={year} data-year={year}>{year}</option>
					))}
				</select>
			</div>
		);
	}

	renderIndicator() {
		const {indicator, currentYear} = this.props;

		if (!indicator) {
			return null;
		}

		return (
			<div className='indicator'>
				<Helmet>;
					<title>{getPageTitle(`${indicator.name} ${currentYear ? `- ${currentYear}` : ''}`)}</title>
				</Helmet>
				<h1>{indicator.name}</h1>
				<p>{indicator.sourceNote !== 'NULL' && indicator.sourceNote}</p>
				{indicator.sourceOrganization && <p>Source: {indicator.sourceOrganization}</p>}
				{this.rendersYears()}

				<IndicatorDataByYearProvider year={currentYear} indicatorId={indicator.id} render={({loading, indicatorDataByYear}) =>
					<IndicatorData currentYear={currentYear} indicator={indicator} loading={loading} data={indicatorDataByYear}/>
				}/>
			</div>
		);
	}

	render() {
		const {loading} = this.props;
		const classNames = classnames('indicator', {'loading': loading});

		return (
			<div className={classNames}>
				{loading ? <Loader/> : this.renderIndicator()}
			</div>
		);
	}
}
