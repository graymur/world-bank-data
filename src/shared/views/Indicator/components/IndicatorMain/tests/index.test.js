/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import {MockedProvider} from 'react-apollo/test-utils';
import IndicatorMain from '../';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const props = {
	match: {
		params: {indicatorId: '1'}
	},
	indicator: {
		id: '1',
		name: 'Indicator name',
		sourceNote: 'Source note',
		sourceOrganization: 'Source organization',
	},
	currentYear: 2000,
	years: [2000, 2001],
	suggestData: {}
};

const getTree = (props, result = {}) => (
	<MockedProvider mocks={[
		{request: {query: '', variables: {cache: false}}, result: {data: result}}
	]}>
		<IndicatorMain {...props}/>
	</MockedProvider>
)

test('Renders preloader', () => {
	const thisProps = Object.assign({}, props);
	thisProps.loading = true;
	const component = renderer.create(getTree(thisProps));

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Renders indicator info', () => {
	const component = renderer.create(getTree(props, {}));

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Handles year change', () => {
	const changeYear = jest.fn();

	const component = shallow(<IndicatorMain {...props} changeYear={changeYear}/>);
	component.find('#year').simulate('change', {target: {value: 2000}});

	expect(changeYear.mock.calls.length).toBe(1);
	expect(changeYear.mock.calls[0][0]).toBe(2000);
});
