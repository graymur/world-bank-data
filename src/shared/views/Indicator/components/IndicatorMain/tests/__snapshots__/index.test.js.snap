// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders indicator info 1`] = `
<div
  className="indicator"
>
  <div
    className="indicator"
  >
    <h1>
      Indicator name
    </h1>
    <p>
      Source note
    </p>
    <p>
      Source: 
      Source organization
    </p>
    <div
      className="indicator__years"
    >
      <select
        className="styled-select"
        defaultValue={2000}
        id="year"
        onChange={[Function]}
      >
        <option
          hidden={true}
          value=""
        >
          Select year
        </option>
        <option
          data-year={2000}
        >
          2000
        </option>
        <option
          data-year={2001}
        >
          2001
        </option>
      </select>
    </div>
    <div
      className="indicator__data loading"
    >
      <span
        className="loader loader"
        style={
          Object {
            "border": "10px solid #f3f3f3",
            "borderTop": "10px solid #3498db",
            "height": "50px",
            "width": "50px",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Renders preloader 1`] = `
<div
  className="indicator loading"
>
  <span
    className="loader"
    style={
      Object {
        "border": "10px solid #f3f3f3",
        "borderTop": "10px solid #3498db",
        "height": "50px",
        "width": "50px",
      }
    }
  />
</div>
`;
